src/systemd/sd-bus-vtable.h

```diff
@@ -90,7 +92,64 @@ struct sd_bus_vtable {
                 },                                                      \
         }
 
-#define SD_BUS_METHOD_WITH_OFFSET(_member, _signature, _result, _handler, _offset, _flags)   \
+/* series of macros to define argument names */
+#define _SD_BUS_ARGSTR_FIRST(a, ...) a
+#define _SD_BUS_ARGSTR_SECOND(a, b, ...) b
+
+#define _SD_BUS_ARGSTR_EMPTY()
+
+#define _SD_BUS_ARGSTR_EVAL(...) _SD_BUS_ARGSTR_EVAL1024(__VA_ARGS__)
+#define _SD_BUS_ARGSTR_EVAL1024(...) _SD_BUS_ARGSTR_EVAL512(_SD_BUS_ARGSTR_EVAL512(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL512(...) _SD_BUS_ARGSTR_EVAL256(_SD_BUS_ARGSTR_EVAL256(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL256(...) _SD_BUS_ARGSTR_EVAL128(_SD_BUS_ARGSTR_EVAL128(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL128(...) _SD_BUS_ARGSTR_EVAL64(_SD_BUS_ARGSTR_EVAL64(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL64(...) _SD_BUS_ARGSTR_EVAL32(_SD_BUS_ARGSTR_EVAL32(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL32(...) _SD_BUS_ARGSTR_EVAL16(_SD_BUS_ARGSTR_EVAL16(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL16(...) _SD_BUS_ARGSTR_EVAL8(_SD_BUS_ARGSTR_EVAL8(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL8(...) _SD_BUS_ARGSTR_EVAL4(_SD_BUS_ARGSTR_EVAL4(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL4(...) _SD_BUS_ARGSTR_EVAL2(_SD_BUS_ARGSTR_EVAL2(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL2(...) _SD_BUS_ARGSTR_EVAL1(_SD_BUS_ARGSTR_EVAL1(__VA_ARGS__))
+#define _SD_BUS_ARGSTR_EVAL1(...) __VA_ARGS__
+
+#define _SD_BUS_ARGSTR_DEFER2(m) m _SD_BUS_ARGSTR_EMPTY _SD_BUS_ARGSTR_EMPTY()()
+
+#define _SD_BUS_ARGSTR_IS_PROBE(...) _SD_BUS_ARGSTR_SECOND(__VA_ARGS__, 0)
+#define _SD_BUS_ARGSTR_PROBE() ~, 1
+
+#define _SD_BUS_ARGSTR_CAT(a,b) a ## b
+
+#define _SD_BUS_ARGSTR_NOT(x) _SD_BUS_ARGSTR_IS_PROBE(_SD_BUS_ARGSTR_CAT(_SD_BUS_ARGSTR__NOT_, x))
+#define _SD_BUS_ARGSTR__NOT_0 _SD_BUS_ARGSTR_PROBE()
+
+#define _SD_BUS_ARGSTR_BOOL(x) _SD_BUS_ARGSTR_NOT(_SD_BUS_ARGSTR_NOT(x))
+
+#define _SD_BUS_ARGSTR_IF_ELSE(condition) _SD_BUS_ARGSTR__IF_ELSE(_SD_BUS_ARGSTR_BOOL(condition))
+#define _SD_BUS_ARGSTR__IF_ELSE(condition) _SD_BUS_ARGSTR_CAT(_SD_BUS_ARGSTR__IF_, condition)
+
+#define _SD_BUS_ARGSTR__IF_1(...) __VA_ARGS__ _SD_BUS_ARGSTR__IF_1_ELSE
+#define _SD_BUS_ARGSTR__IF_0(...)             _SD_BUS_ARGSTR__IF_0_ELSE
+
+#define _SD_BUS_ARGSTR__IF_1_ELSE(...)
+#define _SD_BUS_ARGSTR__IF_0_ELSE(...) __VA_ARGS__
+
+#define _SD_BUS_ARGSTR_HAS_ARGS(...) _SD_BUS_ARGSTR_BOOL(_SD_BUS_ARGSTR_FIRST(_SD_BUS_ARGSTR__END_OF_ARGUMENTS_ __VA_ARGS__)())
+#define _SD_BUS_ARGSTR__END_OF_ARGUMENTS_() 0
+
+#define _SD_BUS_ARGSTR_MAP(m, first, ...)                              \
+  m(first)                                                             \
+  _SD_BUS_ARGSTR_IF_ELSE(_SD_BUS_ARGSTR_HAS_ARGS(__VA_ARGS__))(        \
+    _SD_BUS_ARGSTR_DEFER2(_SD_BUS_ARGSTR__MAP)()(m, __VA_ARGS__)       \
+  )(                                                                   \
+    /* Do nothing, just terminate */                                   \
+  )
```

Comment:
in macros too, please indent by 8ch... (that said, as discussed, maybe let's not do this complex macro)